### PR TITLE
chore(internal): avoid freezegun in ddtrace

### PR DESCRIPTION
**Pull request by romainkomorndatadog**
`freezegun` (and `pytest-freezegun`) may cause wrong durations to be reported when used, so we add an integration that automatically enforces that `ddtrace` be an ignored module by:
- calling `freezegun.configure()` with `extend_ignore_list` to add `ddtrace`
- wrapping `freezegun.configure()` so that any attempt to reset the `default_ignore_list` includes `ddtrace`

Additionally, since `pytest` does not patch all integrations by default, and because we don't want to force users to use `--ddtrace-patch-all` for `freezegun` to be patched, we update our `pytest`  plugin(s) to patch `freezegun` during the `pytest_load_initial_conftests` hook (which should be early enough).

This is not considered a fix as there are no active issues reported and this (so far) primarily affects telemetry recorded by the CI Visibility service.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/__init__.py [hunk 1]
```diff
@@ -36,6 +39,14 @@
     warnings.simplefilter("ignore")
     from .tracer import Tracer as _
 
+# freezegun may cause strange durations to be reported for spans or stopwatch uses
+try:
+    import freezegun
+
+    freezegun.configure(extend_ignore_list=("ddtrace",))
+except ImportError:
+    pass
```
[the remark this answers is not in the document]
Comment (P403n1x87) [reply ddtrace/__init__.py:44]: If we really need to do this, we should consider making this an import hook instead, so we don't force-import `freezegun`

**Review by P403n1x87 — COMMENTED** [review]
My initial reaction is that we should add `freezegun` alongside `gevent` as trigger for module cloning. This at least as a short-term solution. Long-term we could consider having our native timing module/utilities to separate more from the runtime.

**Review by emmettbutler — COMMENTED** [review]
I second @P403n1x87's comment, let's avoid importing freezegun.

File: ddtrace/contrib/freezegun/__init__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Enabling
+~~~~~~~~
+The freezegun integration is enabled by default. Use :func:`patch()<ddtrace.patch>` to enable the integration:
+    from ddtrace import patch
+    patch(freezegun=True)
+
+
+Configuration
+~~~~~~~~~~~~~
+The freezegun integration is not configurable, but may be disabled using DD_PATCH_MODULES:freezegun=false.
```
Comment (vitor-de-araujo) [inline ddtrace/contrib/freezegun/__init__.py:13]: Isn't it like this?
```suggestion
The freezegun integration is not configurable, but may be disabled using DD_PATCH_MODULES=freezegun:false.
```

**Review by vitor-de-araujo — APPROVED** [review]

File: ddtrace/contrib/freezegun/__init__.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+"""
+The freezegun integration reconfigures freezegun's default ignore list to ignore ddtrace.
+
+Enabling
+~~~~~~~~
+The freezegun integration is enabled by default. Use :func:`patch()<ddtrace.patch>` to enable the integration:
```
Comment (romainkomorndatadog) [inline ddtrace/contrib/freezegun/__init__.py:6]:
```suggestion
The freezegun integration is enabled by default. Use :func:`patch()<ddtrace.patch>` to enable the integration::
```

File: ddtrace/contrib/pytest/_plugin_v2.py [hunk 4]
```diff
@@ -99,7 +100,7 @@ def _handle_itr_should_skip(item, test_id) -> bool:
             InternalTest.mark_itr_forced_run(test_id)
             return False
 
-        InternalTest.mark_itr_skipped(test_id)
+        # InternalTest.mark_itr_skipped(test_id)
```
Comment (romainkomorndatadog) [inline ddtrace/contrib/pytest/_plugin_v2.py:103]:
```suggestion
        InternalTest.mark_itr_skipped(test_id)
```

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**